autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/FlowCommunicator.java

```diff
@@ -0,0 +1,44 @@
+package com.sequenceiq.periscope.monitor.handler;
+
+import static java.util.stream.Collectors.toMap;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.client.CloudbreakInternalCrnClient;
+import com.sequenceiq.flow.api.model.FlowCheckResponse;
+
+@Service
+public class FlowCommunicator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FlowCommunicator.class);
+
+    private static final Integer PAGE_SIZE = 50;
+
+    private static final Integer PAGE = 0;
+
+    @Inject
+    private CloudbreakInternalCrnClient internalCrnClient;
+
+    public Map<Long, FlowCheckResponse> getFlowStatusFromFlowIds(Map<String, Long> activityIdsByFlowIds) {
+        LOGGER.info("Sending request to fetch FlowStatus of {} flowIds", activityIdsByFlowIds.size());
+        List<String> flowIds = activityIdsByFlowIds.keySet().stream().limit(PAGE_SIZE).collect(Collectors.toList());
```

Comment:
Limited to 50. Was this the bit that needs to change once pagination is supported ?

Reply:
Yes, this will require a refactor once the API supports pagination. Currently, I am fetching the max allowed elements into a single page and populating that into a list